docs/site/Loopback-component-authentication.md

```diff
@@ -608,6 +608,201 @@ export class MyApplication extends BootMixin(
 }
 ```
 
+## Managing Custom Authentication Strategy Options
+
+This is an **optional** step.
+
+If your custom authentication strategy doesn't require special options, you can
+skip this section.
+
+As previously mentioned in the
+[Using the Authentication Decorator](#using-the-authentication-decorator)
+section, a custom authentication strategy should avoid repeatedly specifying its
+**default** options in the **@authenticate** decorator. Instead, it should
+define its **default** options in one place, and only specify **overriding**
+options in the **@authenticate** decorator when necessary.
+
+Here are the steps for accomplishing this.
+
+### Define the Options Interface and Binding Key
+
+Define an options interface and a binding key for the default options of that
+specific authentication strategy.
+
+```ts
+export interface AuthenticationStrategyOptions {
+  [property: string]: any;
+}
+
+export namespace BasicAuthenticationStrategyBindings {
+  export const DEFAULT_OPTIONS = BindingKey.create<
+    AuthenticationStrategyOptions
+  >('authentication.strategies.basic.defaultoptions');
+}
+```
+
+### Bind the Default Options
+
+Bind the **default** options of the custom authentication strategy to the
+application `application.ts` via the
+`BasicAuthenticationStrategyBindings.DEFAULT_OPTIONS` binding key.
+
+In this hypothetical example, our custom authentication strategy has a
+**default** option of `gatherStatistics` with a value of `true`. (In a real
+custom authentication strategy, the number of options could be more numerous)
+
+```ts
+export class MyApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options?: ApplicationConfig) {
+    super(options);
+
+    //...
+    this.bind(BasicAuthenticationStrategyBindings.DEFAULT_OPTIONS).to({
```

Comment:
We should start to use `this.configure(...)` more.

Reply:
See https://loopback.io/doc/en/lb4/Context.html#configuration-by-convention

Reply:
> We should start to use `this.configure(...)` more.

Hi Raymond, in my other PRs for authentication documentation (Authentication, and Shopping Cart), I don't use yet use `this.configure`. For this PR, since it is a new section in the Authentication document, I want to keep things consistent and provide similar examples with this.bind.

Reply:
Maybe we can create a story to follow-up on changing configuration of bindings to use the new way in not just this package but other places too?